contracts/peripheral/grants/AuraArbBalGrant.sol

```diff
@@ -0,0 +1,251 @@
+       Modifiers 
+    ---------------------------------------------------------------- */
+
+    /**
+     * @notice Modifier that allows only Project or Balancer can trigger a function
+     */
+    modifier onlyAuth() {
+        require(msg.sender == project || msg.sender == balancer, "!auth");
+        _;
+    }
+
+    /**
+     * @notice Modifier that allows only Balancer to trigger a function
+     */
+    modifier onlyBalancer() {
+        require(msg.sender == balancer, "!balancer");
+        _;
+    }
+
+    /**
+     * @notice Modifier that only allows something to be called when the contract is inactive
+     */
+    modifier whileInactive() {
+        require(!active, "active");
+        _;
+    }
+
+    /**
+     * @notice Modifier that only allows something to be called when the contract is active
+     */
+    modifier whileActive() {
+        require(active, "!active");
+        _;
+    }
+
+    /* ----------------------------------------------------------------
+       Init 
+    ---------------------------------------------------------------- */
+
+    /**
+     * @dev Initialise the contract values
+     * @param _aura        AURA token
+     * @param _bpt         BPT token
+     * @param _poolId      poolID of the 8020 pool
+     * @param _tokenOrder  Order of AURA, BAL, ARB in the pool
+     */
+    function init(
+        IERC20 _aura,
+        IERC20 _bpt,
+        bytes32 _poolId,
+        uint256[3] memory _tokenOrder
+    ) external onlyAuth {
+        require(address(AURA) == address(0), "already initialized");
+
+        AURA = _aura;
+        BPT = _bpt;
+        POOL_ID = _poolId;
+
+        for (uint256 i; i < 3; i++) {
+            tokenOrder[i] = _tokenOrder[i];
+        }
+
+        active = true;
+
+        _aura.safeApprove(address(BALANCER_VAULT), type(uint256).max);
+        BAL.safeApprove(address(BALANCER_VAULT), type(uint256).max);
+        ARB.safeApprove(address(BALANCER_VAULT), type(uint256).max);
+    }
+
+    /* ----------------------------------------------------------------
+       Shared Functions
+    ---------------------------------------------------------------- */
+
+    /**
+     * @notice Sends BAL and ARB to balancer and AURA to project
+     * @dev grant must be inactive in order for this to be called
+     */
+    function withdrawBalances() external onlyAuth whileInactive {
+        // Send AURA to project msig
+        AURA.safeTransfer(project, AURA.balanceOf(address(this)));
+        // Send BAL and ARB to balancer msig
+        BAL.safeTransfer(balancer, BAL.balanceOf(address(this)));
+        ARB.safeTransfer(balancer, ARB.balanceOf(address(this)));
+    }
+
+    /**
+     * @notice Join the pool
+     * @dev Only callable by an authenticated party
+     * @dev Only callable when active
+     * @param _minAmountOut Min amount of BPT to get out
+     */
+    function join(uint256 _minAmountOut) external onlyAuth whileActive {
+        _joinPool(_minAmountOut);
```

Comment:
Is there no intention to stake bpt in aura or in a balancer gauge? Could add a small bit of logic that reinvests aura and bal back into the pool to grow it.

Reply:
Not that I am aware, the spec is just that this contract holds the BPT.